src/twisted/test/test_ftp.py

```diff
@@ -11,6 +11,7 @@
 import getpass
 import string
 import random
+import socket
```

Comment:
For what it's worth, these tests are in the old, bad style.  The new, good style would be to consistently use `twisted.internet.testing.MemoryReactor` and not require real, actual IPv6 local connectivity in order to cover this code.  I realize that this isn't really a new problem in this branch, so it doesn't need to be addressed by this branch, but I feel like I should really bring it up to get the word out.